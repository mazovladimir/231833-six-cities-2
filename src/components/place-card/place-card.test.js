import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';

const offer = {image: `image`, priceValue: 100, priceText: `text`, name: `name`, type: `type`};

it(`renders correct PlaceCard component`, () => {
  const mouseOverHandler = jest.fn();
  const mouseOutHandler = jest.fn();
  const component = renderer.create(<PlaceCard offer={offer} mouseOverHandler={mouseOverHandler} mouseOutHandler={mouseOutHandler}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
