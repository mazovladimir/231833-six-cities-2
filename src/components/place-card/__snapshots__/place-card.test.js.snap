// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct PlaceCard component 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="image"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          text
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xmlnsXlink="#icon-bookmark-active"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      type
    </p>
  </div>
</article>
`;

exports[`renders correct PlaceCard component 2`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="image"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          text
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xmlnsXlink="#icon-bookmark-active"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      type
    </p>
  </div>
</article>
`;

exports[`renders correct PlaceCard component 3`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="image"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          text
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xmlnsXlink="#icon-bookmark-active"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      type
    </p>
  </div>
</article>
`;
